src/main/java/net/javadiscord/javabot/systems/starboard/StarboardManager.java

```diff
@@ -216,11 +225,34 @@ private boolean removeMessageFromStarboard(long messageId, MessageChannel channe
 
 	private @NotNull MessageEmbed buildStarboardEmbed(@NotNull Message message) {
 		User author = message.getAuthor();
-		return new EmbedBuilder()
+		EmbedBuilder builder = new EmbedBuilder()
 				.setAuthor(UserUtils.getUserTag(author), message.getJumpUrl(), author.getEffectiveAvatarUrl())
 				.setColor(Responses.Type.DEFAULT.getColor())
 				.setDescription(message.getContentRaw())
-				.setFooter("#" + message.getChannel().getName())
-				.build();
+				.setFooter("#" + message.getChannel().getName());
+		
+		boolean attachedImage = false;
+		
+		List<MessageEmbed> embeds = message.getEmbeds();
+		if(embeds.size() > 0) {
+			MessageEmbed firstEmbed = embeds.get(0);
+			for(Field field : firstEmbed.getFields()) {
+				builder.addField(field);
+			}
+			ImageInfo image = firstEmbed.getImage();
+			if(image != null && image.getUrl() != null) {
+				builder.setImage(image.getUrl());
+				attachedImage = true;
+			}
+		}
+		
+		if(!attachedImage) {
+			List<Attachment> attachments = message.getAttachments();
+			if(attachments.size() == 1) {
+				builder.setImage(attachments.get(0).getUrl());
```

Comment:
Doesn't that result in the image being present twice (once as a normal attachment and once as an embed attachment) if you don't use an `attachment://` URL?

Reply:
It won't. That's why I added a check [here](https://github.com/Defective4/JavaBot/blob/60529d695b853cf1663b6f784eee3ff2421bb921/src/main/java/net/javadiscord/javabot/systems/starboard/StarboardManager.java#L146)
The image won't  attached if the original message has only one attachment

Reply:
I see. Yeah, that would break if there is a single normal attachment and an embed attachment.

Reply:
> and an embed attachment.

Yeah, only if the embed also has an image but yes, I'm aware of that and I don't know how to work around that

Reply:
You could just use an `attachment://` URL (`"attachment://"+fileUpload.getName()`) or don't skip adding the attachment in case `attachedImage` is true (parts of that code could be moved up before the reupload step)